src/Seq.Api/Api/SeqConnection.cs

```diff
@@ -97,16 +100,12 @@ public WatchesResourceGroup Watches
 
         public async Task<ResourceGroup> LoadResourceGroupAsync(string name)
         {
-            ResourceGroup loaded;
-            if (_resourceGroups.TryGetValue(name, out loaded))
-                return loaded;
-
-            if (_root == null)
-                _root = await _client.GetRootAsync();
+            return await _resourceGroups.GetOrAdd(name, ResourceGroupFactory);
```

Comment:
For each SeqConnection instance, ResourceGroupFactory is called once per resource name.
After the task returned by ResourceGroupFactory completes, awaits will just return the value produced by the task.